google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryTest.java

```diff
@@ -392,4 +389,432 @@ public void testOrderByEquality()
+    // TODO(MIEQ): Enable this test against production when possible.
+    assumeTrue(
+        "Skip this test if running against production because order-by-equality is "
+            + "not supported yet.",
+        isRunningAgainstFirestoreEmulator(firestore));
+
+    CollectionReference collection =
+        testCollectionWithDocs(
+            map(
+                "doc1", nestedObject(400),
+                "doc2", nestedObject(200),
+                "doc3", nestedObject(100),
+                "doc4", nestedObject(300)));
+
+    Query query1 =
+        collection
+            .whereLessThanOrEqualTo("metadata.createdAt", 500)
+            .whereGreaterThan("metadata.createdAt", 100)
+            .whereNotEqualTo("name", "room 200")
+            .orderBy("name");
+    checkQuerySnapshotContainsDocuments(query1, "doc4", "doc1");
+
+    Query query2 =
+        collection
+            .whereGreaterThanOrEqualTo("field", "field 100")
+            .whereNotEqualTo(FieldPath.of("field.dot"), 300)
+            .whereLessThan("field\\slash", 400)
+            .orderBy("name", Direction.DESCENDING);
+    checkQuerySnapshotContainsDocuments(query2, "doc2", "doc3");
+
+    // Cursor will add implicit order by fields in the same order as server does.
+    DocumentSnapshot docSnap = collection.document("doc2").get().get();
+    Query query3 =
+        collection
+            .whereGreaterThanOrEqualTo("field", "field 100")
+            .whereNotEqualTo(FieldPath.of("field.dot"), 300)
+            .whereLessThan("field\\slash", 400)
+            .orderBy("name", Direction.DESCENDING)
+            .startAfter(docSnap);
+    checkQuerySnapshotContainsDocuments(query3, "doc3");
+  }
+
+  @Test
+  public void multipleInequalityWithCompositeFilters() throws Exception {
+    // TODO(MIEQ): Enable this test against production when possible.
+    assumeTrue(
+        "Skip this test if running against production because order-by-equality is "
+            + "not supported yet.",
+        isRunningAgainstFirestoreEmulator(firestore));
+
+    CollectionReference collection =
+        testCollectionWithDocs(
+            map(
+                "doc1",
+                map("key", "a", "sort", 0, "v", 5),
+                "doc2",
+                map("key", "aa", "sort", 4, "v", 4),
+                "doc3",
+                map("key", "c", "sort", 3, "v", 3),
+                "doc4",
+                map("key", "b", "sort", 2, "v", 2),
+                "doc5",
+                map("key", "b", "sort", 2, "v", 1),
+                "doc6",
+                map("key", "b", "sort", 0, "v", 0)));
+
+    Query query1 =
+        collection.where(
+            Filter.or(
+                Filter.and(Filter.equalTo("key", "b"), Filter.lessThanOrEqualTo("sort", 2)),
+                Filter.and(Filter.notEqualTo("key", "b"), Filter.greaterThan("v", 4))));
+    // Implicitly ordered by: 'key' asc, 'sort' asc, 'v' asc, __name__ asc
```

Comment:
Similar to Node, please duplicate these tests (one set of tests without cursor, and one set with cursor).